rmgpy/cantherm/molepro.py

```diff
@@ -42,7 +45,197 @@ class MoleProLog:
     
     def __init__(self, path):
         self.path = path
-            
+
+    def getNumberOfAtoms(self):
+        """
+        Return the number of atoms in the molecular configuration used in
+        the MolPro log file.
+        """
+
+        Natoms = 0
+        # Open Molpro log file for parsing
+        f = open(self.path, 'r')
+        line = f.readline()
+        while line != '' and Natoms == 0:
+            # Automatically determine the number of atoms
+            if 'ATOMIC COORDINATES' in line and Natoms == 0:
+                for i in range(4): line = f.readline()
+                while 'Bond lengths' not in line:
+                    Natoms += 1
+                    line = f.readline()
+            line = f.readline()
+        # Close file when finished
+        f.close()
+        # Return the result
+
+        return Natoms - 1
+
+    def loadForceConstantMatrix(self):
+        """
+        Print the force constant matrix by including the print, hessian command in the input file
+        """
+
+        F = None
+
+        Natoms = self.getNumberOfAtoms()
+        Nrows = Natoms * 3
+
+        f = open(self.path, 'r')
+        line = f.readline()
+        while line != '':
+            # Read force constant matrix
+            if 'Force Constants (Second Derivatives of the Energy) in [a.u.]' in line:
+                F = numpy.zeros((Nrows,Nrows), numpy.float64)
+                for i in range(int(math.ceil(Nrows / 5.0))):
+                    # Header row
+                    line = f.readline()
+                    # Matrix element rows
+                    for j in range(i*5, Nrows):
+                        data = f.readline().split()
+                        for k in range(len(data)-1):
+                            F[j,i*5+k] = float(data[k+1].replace('D', 'E'))
+                            F[i*5+k,j] = F[j,i*5+k]
+                # Convert from atomic units (Hartree/Bohr_radius^2) to J/m^2
+                F *= 4.35974417e-18 / 5.291772108e-11**2
+            line = f.readline()
+        # Close file when finished
+        f.close()
+
+        return F
+
+    def loadGeometry(self):
+        """
+        Return the optimum geometry of the molecular configuration from the
+        Molpro .out file. If multiple such geometries are identified, only the
+        last is returned.
+        """
+
+        symbol = []; coord = []
+
+        f = open(self.path, 'r')
+        line = f.readline()
+        while line != '':
+            # Automatically determine the number of atoms
+            if 'Current geometry' in line:
```

Comment:
Loading the geometry fails when it is read from the freq.log instead of the opt.log (as we advise users to do when using hindered rotors, e.g., CanTherm documentation section 3.4.1.2). We should also be able to parse the Molpro freq.log geometry format:
```
 ATOMIC COORDINATES

 NR  ATOM    CHARGE       X              Y              Z

   1  S      16.00   -9.453662536    4.274754420    0.000189541
   2  H       1.00   -9.898568657    6.789652041   -0.000157075
   3  S      16.00   -5.465181259    4.791095755    0.000139265
   4  H       1.00   -5.020275141    2.276198114    0.000280349

 Bond lengths in Bohr (Angstrom)
```

Reply:
@rgillis8, I remembered what I meant in this comment:
When reading the geometry from a Molpro FREQ file, Cantherm cannot find the `Current geometry` line (see example FREQ geometry above), so it skips it. As a consequence, `symbol` remains an empty list, and `loadGeomerty()` cannot return the correct mass (which is latter required in statmech.py `if F is not None and len(mass) > 1 and len(rotors) > 0:` to calc rotors). Could you take a look at improving `loadGeometry()` for these cases?